subprojects/core/src/integTest/groovy/org/gradle/api/tasks/bundling/ArchiveIntegrationTest.groovy

```diff
@@ -238,6 +238,11 @@ class ArchiveIntegrationTest extends AbstractIntegrationSpec {
             }
 '''
         when:
+        executer.expectDocumentedDeprecationWarning(
+            "Creating a tarTree from a resource without a backing file has been deprecated. " +
+                "This will fail with an error in Gradle 8.0. Use a task or declare a dependency to create the tar file. " +
```

Comment:
"Use a task or declare a dependency to create the tar file"

This is not totally clear for me. What task or dependency would I have created in this example here?

Reply:
IIUC, I could also use `project.resource()` to create the resource, and that'd be fine, no?

Reply:
`project.resource` creates a resource from a file. So where'd you get that file from? For me, the resourceless tar tree is about having something which is not coming from a file. Like an URL. If you have a URL, you would declare an ivy repository to that URL and then declare a dependency on that file at the URL. If you can't do that, then you'd need to create a task which creates the tar file on disk in the way the resource created it before.

IIUC, there is no way to declare a resource without a backing file without creating an implementation of `ReadableResource`. We don't ship with any implementation which does that for you.

How would you phrase that?

Reply:
So it seems that what we want is for the build engineer to turn their resource into a file in some way, and use `tarTree()` on that file, no? I think we should say that, and perhaps give a few examples of how they could do that.

Reply:
I reworded the message.